src/utils/dom.ts

```diff
@@ -162,24 +162,33 @@ interface GetSlottedOptions {
   selector?: string;
 }
 
+const defaultSlotSelector = "> :not([slot])";
+
 export function getSlotted<T extends Element = Element>(
```

Comment:
I think this should be typed like so:

```ts
export function getSlotted<T extends Element = Element>(element: Element, options: GetSlottedOptions & { all: true }): T[];
export function getSlotted<T extends Element = Element>(element: Element, options?: GetSlottedOptions): T[];
export function getSlotted<T extends Element = Element>(element: Element, slotName: string | string[], options: GetSlottedOptions & { all: true }): T[];
export function getSlotted<T extends Element = Element>(element: Element, slotName: string | string[], options?: GetSlottedOptions): T[];
export function getSlotted<T extends Element = Element>(
  element: Element,
  slotNameOrOptions: string | string[] | GetSlottedOptions,
  optionsOrUndefined?: GetSlottedOptions): (T | null) | T[] { /* ... */ }
```

Can you create a separate refactor issue this one and assign it to me?